cxx-qt-build/src/lib.rs

```diff
@@ -55,7 +54,7 @@ pub struct GeneratedCpp {
 
 impl GeneratedCpp {
     /// Generate QObject and cxx header/source C++ file contents
-    pub fn new(rust_file_path: &impl AsRef<std::path::Path>) -> Self {
+    pub fn new(rust_file_path: &impl AsRef<Path>) -> Self {
```

Comment:
Just curious what the advantage of `&impl AsRef<Path>` over `&Path` is when `Path` is already a slice of `PathBuf` ?

Reply:
`&impl AsRef<Path>` allows passing `String` and `&str` so the caller doesn't need to explicitly create a `Path` or `PathBuf`

Reply:
OK, so for public API that makes sense i guess.